pkg/daemon/update.go

```diff
@@ -89,23 +89,25 @@ func (dn *Daemon) updateOSAndReboot(newConfig *mcfgv1.MachineConfig) error {
 			Factor:   2,
 		}
 		var lastErr error
-		wait.ExponentialBackoff(backoff, func() (bool, error) {
+		if err := wait.ExponentialBackoff(backoff, func() (bool, error) {
 			err := drain.Drain(dn.kubeClient, []*corev1.Node{node}, &drain.DrainOptions{
 				DeleteLocalData:    true,
 				Force:              true,
 				GracePeriodSeconds: 600,
 				IgnoreDaemonsets:   true,
 			})
-			if err != nil {
-				glog.Infof("Draining failed with: %v; retrying...", err)
-				lastErr = err
-				return false, nil
+			if err == nil {
+				return true, nil
+			}
+			lastErr = fmt.Errorf("draining failed with: %v", err)
```

Comment:
How about just doing `glog.Infof()` for the error message here and leaving `lastErr = err`? Then if we wrap it later, we can still get back the original error that happened. (Anyway, "failed to drain node (5 tries): draining failed with: ..." is kinda repetitive, no? :))

Reply:
:+1:

Reply:
fixed, thanks!